fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Count.java

```diff
@@ -31,21 +31,28 @@
 public class Count extends AggregateFunction {
 
     private final boolean isStar;
+    private final boolean isDistinct;
```

Comment:
move it to super class, and add a two parameters constructor. the original constructor set it to false by default. In super class, method isDistinct return this value directly